New test all-headers/base/sacado_product_type.h.debug fails

@asartori86 -- I get this error:
```
> ctest -V -R all-headers/base/sacado_product_type.h.debug                                                                                                              
UpdateCTestConfiguration  from :/u/bangerth/p/deal.II/2/build/DartConfiguration.tcl                                                                                               
UpdateCTestConfiguration  from :/u/bangerth/p/deal.II/2/build/DartConfiguration.tcl                                                                                                    
Test project /u/bangerth/p/deal.II/2/build                                                                                                                                                
Constructing a list of tests                                                                                                                                                                    
Done constructing a list of tests                                                                                                                                                                
Checking test dependency graph...                                                                                                                                                                    
Checking test dependency graph end                                                                                                                                                                      
test 2065
    Start 2065: all-headers/base/sacado_product_type.h.debug

2065: Test command: /usr/local/bin/cmake "-DTRGT=base-sacado_product_type.h.debug.build" "-DTEST=all-headers/base/sacado_product_type.h.debug" "-DBINARY_DIR=/u/bangerth/p/deal.II/2/build/tests/all-headers" "-DGUARD_FILE=/u/bangerth/p/deal.II/2/build/tests/all-headers/base-sacado_product_type.h.debug_interrupt_guard.cc" "-P" "/u/bangerth/p/deal.II/2/build/share/deal.II/scripts/run_test.cmake"
2065: Test timeout computed to be: 600
2065: Test all-headers/base/sacado_product_type.h.debug: BUILD
2065: ===============================   OUTPUT BEGIN  ===============================
2065: all-headers/base/sacado_product_type.h.debug: BUILD failed. Output:
2065: [1/2] Building CXX object CMakeFiles/base-sacado_product_type.h.debug.dir/test_header.cc.o
2065: FAILED: /usr/bin/c++   -DDEBUG -DHEADER="<deal.II/base/sacado_product_type.h>" -I/u/bangerth/p/deal.II/2/build/include -I/u/bangerth/p/deal.II/2/dealii/include -I/u/bangerth/p/deal.II/2/dealii/bundled/umfpack/UMFPACK/Include -I/u/bangerth/p/deal.II/2/dealii/bundled/umfpack/AMD/Include -I/u/bangerth/p/deal.II/2/dealii/bundled/boost-1.56.0/include -I/u/bangerth/p/deal.II/2/dealii/bundled/muparser_v2_2_4/include -I/apps/openmpi/1.6.5/include -I/w/bangerth/share/software/trilinos-11.10.2-mpi/include -I/w/bangerth/share/software/petsc-3.4.2-mpi/include -I/w/bangerth/share/software/petsc-3.4.2-mpi/x86_64/include -I/w/bangerth/share/software/p4est-1.1/FAST/include    -pedantic -fpic -Wall -Wextra -Wpointer-arith -Wwrite-strings -Wsynth -Wsign-compare -Wswitch -Woverloaded-virtual -Wno-long-long  -std=c++0x -B/w/bangerth/share/software/bin/ld -O0 -ggdb -Wa,--compress-debug-sections -MMD -MT CMakeFiles/base-sacado_product_type.h.debug.dir/test_header.cc.o -MF "CMakeFiles/base-sacado_product_type.h.debug.dir/test_header.cc.o.d" -o CMakeFiles/base-sacado_product_type.h.debug.dir/test_header.cc.o -c /u/bangerth/p/deal.II/2/dealii/tests/all-headers/test_header.cc
2065: In file included from /u/bangerth/p/deal.II/2/dealii/tests/all-headers/test_header.cc:16:0:
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:89:23: error: ‘SymmetricTensor’ does not name a type
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:89:23: error: ISO C++ forbids declaration of ‘parameter’ with no type [-fpermissive]
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:89:38: error: expected ‘,’ or ‘...’ before ‘<’ token
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h: In function ‘Sacado::Fad::DFad<T> dealii::scalar_product(int)’:
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:95:12: error: ‘t1’ was not declared in this scope
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:95:23: error: ‘t2’ was not declared in this scope
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h: At global scope:
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:112:23: error: ‘Tensor’ does not name a type
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:112:23: error: ISO C++ forbids declaration of ‘parameter’ with no type [-fpermissive]
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:112:29: error: expected ‘,’ or ‘...’ before ‘<’ token
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:112:1: error: redefinition of ‘template<int dim, class Number, class T> Sacado::Fad::DFad<T> dealii::scalar_product(int)’
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:89:1: error: ‘template<int dim, class Number, class T> Sacado::Fad::DFad<T> dealii::scalar_product(int)’ previously declared here
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h: In function ‘Sacado::Fad::DFad<T> dealii::scalar_product(int)’:
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:115:25: error: ‘t2’ was not declared in this scope
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:115:29: error: ‘t1’ was not declared in this scope
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h: At global scope:
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:131:23: error: ‘SymmetricTensor’ does not name a type
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:131:23: error: ISO C++ forbids declaration of ‘parameter’ with no type [-fpermissive]
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:131:38: error: expected ‘,’ or ‘...’ before ‘<’ token
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:131:1: error: redefinition of ‘template<int dim, class Number, class T> Sacado::Fad::DFad<T> dealii::scalar_product(int)’
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:89:1: error: ‘template<int dim, class Number, class T> Sacado::Fad::DFad<T> dealii::scalar_product(int)’ previously declared here
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h: In function ‘Sacado::Fad::DFad<T> dealii::scalar_product(int)’:
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:137:12: error: ‘t1’ was not declared in this scope
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:137:23: error: ‘t2’ was not declared in this scope
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h: At global scope:
2065: /u/bangerth/p/deal.II/2/dealii/include/deal.II/base/sacado_product_type.h:154:23: error: ‘Tensor’ does not name a type
...
```

I think the header file is simply missing a couple of `#include` statements to be self-contained. Can you add them?

--- a/include/deal.II/base/sacado_product_type.h
+++ b/include/deal.II/base/sacado_product_type.h
@@ -17,6 +17,7 @@
 #define dealii__sacado_product_type_h
 
 #include <deal.II/base/config.h>
+#include <deal.II/base/tensor.h>
 #include <deal.II/base/template_constraints.h>
 
 #ifdef DEAL_II_WITH_TRILINOS

--- a/include/deal.II/base/sacado_product_type.h
+++ b/include/deal.II/base/sacado_product_type.h
@@ -18,6 +18,7 @@
 
 #include <deal.II/base/config.h>
 #include <deal.II/base/tensor.h>
+#include <deal.II/base/symmetric_tensor.h>
 #include <deal.II/base/template_constraints.h>
 
 #ifdef DEAL_II_WITH_TRILINOS

